### PR TITLE
feat: Simplify 'Ctrl+S' Functionality in Editor to Prevent Default Save Prompt

### PR DESCRIPTION
**Pull request by Jacksonmills**
- Add a save handler that listens for Ctrl + S key combination and displays a toast notification with the title "Saved" and the description "Your code has been saved" when the current wrapper ref has focus

## Description
Pressing 'Ctrl+S' in the code editor triggers the browser's default save function. This behavior is counterintuitive for users accustomed to IDEs where 'Ctrl+S' typically saves progress within the app or performs another action, such as running tests. Disabling this or binding 'Ctrl+S' to a more useful function, like running tests, would enhance usability.

## Related Issue
Closes [#1050](https://github.com/typehero/typehero/issues/1050)

## Motivation and Context
This change would streamline the coding experience and align the editor's functionality with common developer practices, improving the platform's overall user experience.

## How Has This Been Tested?
Tested locally, I can add more if requested!
[ ] - Need to test mac `cmd+s`

## Screenshots/Video (if applicable):
https://github.com/typehero/typehero/assets/19780885/9e9e9847-6192-4741-800f-6deefc342d04

### CODE REVIEW

File: packages/monaco/src/split-editor.tsx [hunk 1]
```diff
@@ -89,6 +91,50 @@ export default function SplitEditor({
   const monacoRef = useRef<typeof import('monaco-editor')>();
   const editorRef = useRef<monacoType.editor.IStandaloneCodeEditor>();
 
+  const [wrapperFocused, setWrapperFocused] = useState(false);
```
Comment (Dun-sin) [inline packages/monaco/src/split-editor.tsx:94]: quick question, why is it called wrapperFocused, i think the naming can be better for readability. just a thought.

Comment (Jacksonmills) [reply packages/monaco/src/split-editor.tsx:94]: in this context the wrapper is just the wrapper div we have reference to with useRef, I think if we wanted to clarify I could rename to `isWrapperInnerFocused`

File: packages/monaco/src/split-editor.tsx [hunk 2]
```diff
@@ -89,6 +91,49 @@ export default function SplitEditor({
...
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!isWrapperInnerFocused) {
+      return;
+    }
+
+    const saveHandler = (e: KeyboardEvent) => {
+      if (e.ctrlKey && e.code === 'KeyS') {
```
Comment (bautistaaa) [inline packages/monaco/src/split-editor.tsx:120]: cant we determine if the wrapper is in focus during this effect with its ref vs introducing a piece of state and another effect?
 
using ref.current.activeElement or hasFocus. cant remember the apis

Comment (Jacksonmills) [reply packages/monaco/src/split-editor.tsx:120]: I will try that sure!

**Comment by Mahmoudgalalz** [comment]
also, show a toast that says "It's already saved in your Localstorage"

**Comment by Mahmoudgalalz** [comment]
I can test on mac, just finish it

**Comment by Jacksonmills** [comment]
> also, show a toast that says "It's already saved in your Localstorage"

hm I am not sure if thats the best copy imo

**Comment by bautistaaa** [comment]
> > also, show a toast that says "It's already saved in your Localstorage"
> 
> hm I am not sure if thats the best copy imo

it’s fine as is

**Review by Mahmoudgalalz — COMMENTED** [review]
will test on my macbook

**Review by Mahmoudgalalz — APPROVED** [review]
yeah, this works fine on mac, I just added it a min ago to fix it

**Review by bautistaaa — APPROVED** [review]

File: packages/monaco/src/split-editor.tsx (hunk 1 above)
Comment (Dun-sin) [reply packages/monaco/src/split-editor.tsx]: since it's a boolean i would say it's a better and more expressive name

**Comment by bautistaaa** [comment]
> yeah, this works fine on mac, I just added it a min ago to fix it

can you run pnpm checks:fix youre failing prettier check

**Comment by bautistaaa** [comment]
> > yeah, this works fine on mac, I just added it a min ago to fix it
> 
> can you run pnpm checks:fix youre failing prettier check

nm i got it

**Comment by Jacksonmills** [comment]
> > > yeah, this works fine on mac, I just added it a min ago to fix it
> > 
> > 
> > can you run pnpm checks:fix youre failing prettier check
> 
> nm i got it

Ah sorry about that! Noted for next time 👍

**Pull request merged**